staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -168,6 +169,7 @@ func addDebugFlags(cmd *cobra.Command, opt *DebugOptions) {
 	cmd.Flags().BoolVar(&opt.Replace, "replace", opt.Replace, i18n.T("When used with '--copy-to', delete the original Pod."))
 	cmd.Flags().StringToString("env", nil, i18n.T("Environment variables to set in the container."))
 	cmd.Flags().StringVar(&opt.Image, "image", opt.Image, i18n.T("Container image to use for debug container."))
+	cmd.Flags().BoolVar(&opt.Privileged, "privileged", opt.Privileged, i18n.T("If true, run the container for the node target in privileged mode."))
```

Comment:
Nit1: `target node` or `node target`? I find both in the original code.

https://github.com/kubernetes/kubernetes/blob/2d58b72889b5c617b2c70d2734eef6d391362e1b/staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go#L362-L369
> // visitNode handles debugging for node targets by creating a privileged pod running in the host namespaces.<br class="Apple-interchange-newline">
The debug pod is mounting hostPath using hostNetwork=true/HostIPC=true/HostPID=true. As the comments described, it should be with "Privileged=true" by default.

There should not be `--privileged` flag as there is no flag like `--hostNetwork` or `--hostPID`

Reply:
@pacoxu Thank you for your advice. Setting privileged true by default sounds more reasonable than adding the flag to me. Since a node debug pod already mounts the root directory of the host, it virtually has privileged capabilities in most cases.

For example, we can escalate privileges with host's docker even now.

```
$ kubectl debug node/<NODE> -it --image busybox

// allow us to use host's commands
# chroot /host
# capsh --print
Current: = cap_chown,cap_dac_override,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_net_bind_service,cap_net_raw,cap_sys_chroot,cap_mknod,cap_audit_write,cap_setfcap+eip
Bounding set =cap_chown,cap_dac_override,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_net_bind_service,cap_net_raw,cap_sys_chroot,cap_mknod,cap_audit_write,cap_setfcap
Securebits: 00/0x0/1'b0
 secure-noroot: no (unlocked)
 secure-no-suid-fixup: no (unlocked)
 secure-keep-caps: no (unlocked)
uid=0(root)
gid=0(root)
groups=10(uucp)

// we can esalate privileges with docker's --privileged
# docker run -it --privileged -v /:/host --pid=host --net=host --ipc=host busybox
# chroot /host
# capsh --print
Current: = cap_chown,cap_dac_override,cap_dac_read_search,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_linux_immutable,cap_net_bind_service,cap_net_broadcast,cap_net_admin,cap_net_raw,cap_ipc_lock,cap_ipc_owner,cap_sys_module,cap_sys_rawio,cap_sys_chroot,cap_sys_ptrace,cap_sys_pacct,cap_sys_admin,cap_sys_boot,cap_sys_nice,cap_sys_resource,cap_sys_time,cap_sys_tty_config,cap_mknod,cap_lease,cap_audit_write,cap_audit_control,cap_setfcap,cap_mac_override,cap_mac_admin,cap_syslog,cap_wake_alarm,cap_block_suspend,cap_audit_read+eip
Bounding set =cap_chown,cap_dac_override,cap_dac_read_search,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_linux_immutable,cap_net_bind_service,cap_net_broadcast,cap_net_admin,cap_net_raw,cap_ipc_lock,cap_ipc_owner,cap_sys_module,cap_sys_rawio,cap_sys_chroot,cap_sys_ptrace,cap_sys_pacct,cap_sys_admin,cap_sys_boot,cap_sys_nice,cap_sys_resource,cap_sys_time,cap_sys_tty_config,cap_mknod,cap_lease,cap_audit_write,cap_audit_control,cap_setfcap,cap_mac_override,cap_mac_admin,cap_syslog,cap_wake_alarm,cap_block_suspend,cap_audit_read
Securebits: 00/0x0/1'b0
 secure-noroot: no (unlocked)
 secure-no-suid-fixup: no (unlocked)
 secure-keep-caps: no (unlocked)
uid=0(root)
gid=0(root)
groups=10(uucp)
```

Reply:
@verb Do you have any concerns if we set privileged true for the node debug pod by default?

Reply:
@tksm is that true for any runtime, though? Does mounting host root always allow privilege escalation? Is there a use case for node debugging with a non-root container image?

@tallclair Do you have any suggestions about how to keep `kubectl debug node` useful in clusters with security-focused admission controllers?

Reply:
@verb Thank you! I missed the case with a non-root container.

> is that true for any runtime, though? Does mounting host root always allow privilege escalation?

No, it is not always true. If a user is not allowed to create root containers, for example, by MustRunAsNonRoot in PSP, they cannot escalate their privileges with mounting host root.

In that case, I think adding `--privileged` flag is more flexible than making it default.

Reply:
@pacoxu Fixed privileged flag usage in https://github.com/kubernetes/kubernetes/pull/101897/commits/8b8c4399ccb292c52009311042654f21c461ae5f . Thank you for your suggestion.